src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -241,16 +251,22 @@ def _forward(self, model_inputs):
                     stride = [stride]
 
                 apply_stride(tokens, stride)
+            out = {"tokens": tokens}
         else:
             logger.warning("This is an unknown class, treating it as CTC.")
             outputs = self.model(**model_inputs)
             tokens = outputs.logits.argmax(dim=-1)
-        return {"tokens": tokens, "is_last": is_last}
+            out = {"tokens": tokens}
+        return {"is_last": is_last, **out}
 
     def postprocess(self, model_outputs):
-        skip_special_tokens = False if "CTC" in self.tokenizer.__class__.__name__ else True
-        tokens = np.concatenate([outputs["tokens"].numpy() for outputs in model_outputs], axis=-1)
-        tokens = tokens.squeeze(0)
-
-        recognized_string = self.tokenizer.decode(tokens, skip_special_tokens=skip_special_tokens)
-        return {"text": recognized_string}
+        if self.type == "ctc_with_lm":
+            logits = np.concatenate([outputs["logits"].numpy() for outputs in model_outputs], axis=1)
+            logits = logits.squeeze(0)
+            text = self.tokenizer.decode_beams(logits)[0][0]
```

Comment:
not supper happy that the decoder is now the tokenizer in terms of naming, but I see why it makes sense! Also should we allow the user to pass **kwargs to `decode_beams(...)`?

Reply:
Yes, I hesitated back&forth. Adding a decoder, means adding a bunch of code everywhere to automatically load it up etc.. For a single pretty niche (currently) use case.
Adding it as tokenizer, is defendable as you said, and makes things simpler for all utils. It's not however really a tokenizer (since it takes `logits` not `input_ids`). The good thing, is that for the pipeline it should roughly be seen as an implementation detail, not really something users should depend on  (if it's `tokenizer`).

I don't like adding `kwargs` without careful thought (pipelines are already a kwarg mess, adding on top is something worth at least a pause).

- Isn't there a way to make the decoder itself have the kwargs ? That means better ownership of information. (overridding kwargs means the user now has to declare himself `decoder = pytctc.Decode(..., **kwargs); pipe = pipeline(..., decoder=decoder)` and at least who eats those arguments is extra clear.
- Adding them to the raw pipeline is like the `generate_kwargs`, not every argument can be changed in the future, and user has no idea if the argument was correctly passed. (Misnamed argument gets silently discarded)
- Other option if `decoder` doesn't own it is to have `pipeline(..., decoder_kwargs={..})`. This is cleaner as it adds only 1 kwargs and it's unlikely to be miss attributed. However, those are not "owned" by the decoder itself, meaning the behavior is pipeline specific. Going that route the arguments to pipelines are going to get extremely long (they already are).

Reply:
I moved to `decoder` in the end.